### PR TITLE
feat: add parsing of isFlag property by Examples

### LINKED ISSUE #67: Add parsing of "args" prefix by examples
Sometimes, in **options** section it is no information what can be interpreted that option is not flag. But, in **examples** there is so.

Example:

```
  Options:
    --plugin          Include third-party plugins [Can be set multiple times]

  Examples:
    $ xo --plugin=html --extension=html
```

Please, add parsing of "args" prefix by examples.

### PR DESCRIPTION
**Pull request by ifedchankau**
- Add parsing of argument property isFlag by Examples section

Closes #67

### CODE REVIEW

File: src/sections.js [hunk 1]
```diff
@@ -40,6 +40,31 @@ const examples = (section, context) => {
...
+};
+
+/**
+ * Parse argument property isFlag by examples section
+ * @param {string} section - examples section
+ * @param {object} context - internal config
+ */
+const parseArgs = (section, context) => {
+    section.split('\n').forEach((string) => {
+        const matches = string.match(context.regexp.examplesArgument);
```
Comment (romanrostislavovich) [inline src/sections.js]: We can use global regex with multiline, then we can remove `split` function

File: src/sections.js [hunk 2]
```diff
@@ -40,6 +40,31 @@ const examples = (section, context) => {
...
+    });
+};
+
+/**
+ * Make argument property isFlag false
+ * @param {string} name - name of an argument
+ * @param {object} context - internal config
+ */
+const makeArgNotFlag = (name, context) => {
+    context.options.forEach((option) => {
```
Comment (romanrostislavovich) [inline src/sections.js]: Change `forEach` to `find` for array

File: src/sections.js (hunk 1 above)
Comment (romanrostislavovich) [reply src/sections.js]: I suggest to remove this function to `arguments.js` file

Comment (romanrostislavovich) [reply src/sections.js]: Please use pure function and think out it for this implementation. You can, i know it :)

File: src/arguments.js [hunk 3]
```diff
@@ -135,6 +135,21 @@ const removeExtraEnumValues = (enumArray, regexp) => {
     return result.length !== 0 ? result : null;
 };
 
+/**
+ * Parse argument property isFlag by examples section
+ * @param {string} section - examples section
+ * @param {object} context - internal config
+ */
+const parseIsFlag = (section, context) => {
```
Comment (romanrostislavovich) [inline src/arguments.js]: We need better name for this function

File: src/arguments.js [hunk 4]
```diff
@@ -135,6 +135,21 @@ const removeExtraEnumValues = (enumArray, regexp) => {
...
+    const regularExp = new RegExp(context.regexp.examplesArgument, 'gim');
+    while (match = regularExp.exec(section)) {
+        context.options.find((option) => {
+            return option.longName === match[1] ||
+                option.shortName === match[1];
+        }).isFlag = false;
```
Comment (romanrostislavovich) [inline src/arguments.js]: Check if `find` return `undefined`

File: test/parsing/files/full.example.valid.json [hunk 5]
```diff
@@ -13,8 +13,8 @@
          "type": "object",
          "properties": {
             "--extension": {
-               "id": "--extension",
                "type": null,
+               "id": "args:--extension",
```
Comment (romanrostislavovich) [inline test/parsing/files/full.example.valid.json]: Why for this properties added prefix `args` ?

File: src/arguments.js (hunk 3 above)
Comment (Arcle1) [reply src/arguments.js]: @svoboda-rabstvo yes, maybe you have any ideas?

**Review by romanrostislavovich — COMMENTED** [review]
Check my answer

File: src/arguments.js [hunk 6]
```diff
@@ -135,6 +135,22 @@ const removeExtraEnumValues = (enumArray, regexp) => {
...
+/**
+ * Parse argument property isFlag by examples section
+ * @param {string} section - examples section
+ * @param {object} context - internal config
+ */
+const parseIsFlag = (section, context) => {
+    const regularExp = new RegExp(context.regexp.examplesArgument, 'gim');
+    while (match = regularExp.exec(section)) {
+        let option = context.options.find((option) => {
+            return option.longName === match[1] ||
```
Comment (romanrostislavovich) [inline src/arguments.js]: I suggest add check if `option` is a flag and also we can check option name like a `match[1] === (option.longName || option.shortName)`

File: test/parsing/files/full.example.valid.json (hunk 5 above)
Comment (romanrostislavovich) [reply test/parsing/files/full.example.valid.json]: @Arcle1 ping :)

**Review by romanrostislavovich — APPROVED** [review]
+1

**Pull request merged**